lib/step.js

```diff
@@ -139,20 +139,29 @@ class Step {
     });
   }
 
-  next() {
+  _moveTo(action) {
     Promise.resolve().then(() => {
       if (this.delegate.didFinishStep) {
         return this.delegate.didFinishStep(
           this, this.index, this.tutorial);
       }
     }).then(() => {
-      this.tutorial.next();
+      action();
     }).catch(error => {
       console.log(error);
-      this.tutorial.next();
+      action();
     });
   }
 
+  previous() {
+    let step = this.tutorial.stepNum(this) - 1;
```

Comment:
I used `stepNum()` instead of `this.index` because I had made `tutorial.next()` accepts a one-indexed step number. So, if I change `tutorial.next()` back to zero-indexed then I could use `this.index`. But either way, the `-1` is still needed to get the index of the previous step.